internal/clusterinfo/types.go

```diff
@@ -0,0 +1,286 @@
+	InFlightCount int64           `json:"in_flight_count"`
+	DeferredCount int64           `json:"defer_count"`
+	RequeueCount  int64           `json:"requeue_count"`
+	TimeoutCount  int64           `json:"timeout_count"`
+	MessageCount  int64           `json:"msg_count"`
+	ClientCount   int             `json:"-"`
+	Selected      bool            `json:"-"`
+	NodeStats     []*ChannelStats `json:"nodes"`
+	Clients       []*ClientStats  `json:"clients"`
+	Paused        bool            `json:"paused"`
+
+	E2eProcessingLatency *quantile.E2eProcessingLatencyAggregate `json:"e2e_processing_latency"`
+}
+
+func (c *ChannelStats) Add(a *ChannelStats) {
+	c.Node = "*"
+	c.Depth += a.Depth
+	c.MemoryDepth += a.MemoryDepth
+	c.BackendDepth += a.BackendDepth
+	c.InFlightCount += a.InFlightCount
+	c.DeferredCount += a.DeferredCount
+	c.RequeueCount += a.RequeueCount
+	c.TimeoutCount += a.TimeoutCount
+	c.MessageCount += a.MessageCount
+	c.ClientCount += a.ClientCount
+	if a.Paused {
+		c.Paused = a.Paused
+	}
+	c.NodeStats = append(c.NodeStats, a)
+	sort.Sort(ChannelStatsByHost{c.NodeStats})
+	if c.E2eProcessingLatency == nil {
+		c.E2eProcessingLatency = &quantile.E2eProcessingLatencyAggregate{
+			Addr:    c.Node,
+			Topic:   c.TopicName,
+			Channel: c.ChannelName,
+		}
+	}
+	c.E2eProcessingLatency.Add(a.E2eProcessingLatency)
+}
+
+type ClientStats struct {
+	Node              string        `json:"node"`
+	RemoteAddress     string        `json:"remote_address"`
+	Name              string        `json:"name"` // TODO: deprecated, remove in 1.0
+	Version           string        `json:"version"`
+	ClientID          string        `json:"client_id"`
+	Hostname          string        `json:"hostname"`
+	UserAgent         string        `json:"user_agent"`
+	ConnectTs         int64         `json:"connect_ts"`
+	ConnectedDuration time.Duration `json:"connected"`
+	InFlightCount     int           `json:"in_flight_count"`
+	ReadyCount        int           `json:"ready_count"`
+	FinishCount       int64         `json:"finish_count"`
+	RequeueCount      int64         `json:"requeue_count"`
+	MessageCount      int64         `json:"message_count"`
+	SampleRate        int32         `json:"sample_rate"`
+	Deflate           bool          `json:"deflate"`
+	Snappy            bool          `json:"snappy"`
+	Authed            bool          `json:"authed"`
+	AuthIdentity      string        `json:"auth_identity"`
+	AuthIdentityURL   string        `json:"auth_identity_url"`
+
+	TLS                           bool   `json:"tls"`
+	CipherSuite                   string `json:"tls_cipher_suite"`
+	TLSVersion                    string `json:"tls_version"`
+	TLSNegotiatedProtocol         string `json:"tls_negotiated_protocol"`
+	TLSNegotiatedProtocolIsMutual bool   `json:"tls_negotiated_protocol_is_mutual"`
+}
+
+// UnmarshalJSON does post-processing of ConnectedDuration
+func (s *ClientStats) UnmarshalJSON(b []byte) (err error) {
+	type c ClientStats // used to prevent recursion
+	var i c
```

Comment:
We can't unmarshal directly to something that's a `type` of `ClientStats` because this function **is** the unmarshal code for that (recursion here would be bad); We just have to alias the type to work around that. That's your question right?